cmd/juju/application/storage.go

```diff
@@ -0,0 +1,267 @@
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package application
+
+import (
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/juju/cmd/v3"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"github.com/juju/names/v5"
+
+	"github.com/juju/juju/api/client/application"
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/juju/block"
+	"github.com/juju/juju/cmd/juju/config"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/cmd/output"
+	"github.com/juju/juju/storage"
+)
+
+const (
+	appStorageConfigSummary = "Displays or sets storage constraints values on an application."
+
+	appStorageConfigDoc = `
+To view all storage constraints values for the given application:
+
+    juju application-storage <application>
+
+	By default, the config will be printed in a tabular format. You can instead
+print it in the ` + "`json`" + ` or ` + "`yaml`" + ` format using the ` + "`--format`" + ` flag:
+
+   	juju application-storage <application> --format json
+    juju application-storage <application> --format yaml
+
+To view the value of a single storage key:
+
+    juju application-storage <application> <storage-key>
+
+To set storage constraint values on an application:
+
+    juju application-storage <application> key1=size1, key2=val2 ...
+`
+	appStorageConfigExamples = `
+Print the storage constraints for all keys of the postgresql application:
+
+    juju application-storage postgresql
+
+Print the storage constraints for the pgdata storage key of the postgresql application:
+
+    juju application-storage postgresql pgdata
+
+Set the size to 100GiB, pool name to "rootfs", and count to 1 for the mysql application:
+
+    juju application-storage mysql database=100G,rootfs,1
+
+If no size is provided, Juju uses the minimum size required by the charm. If the charm does not specify a minimum, the default is 1 GiB. 
+This value is then applied when updating the application’s storage.
+
+    juju application-storage mysql database=,rootfs,1
+
+If no pool is provided, Juju selects the default storage pool from the environment.
+This pool will be recorded as the updated value for the application’s storage.
+
+	juju application-storage mysql database=100G,,1
+
+If no count is provided, Juju uses the minimum count required by the charm. 
+That count will be used when updating the application’s storage.
+
+	juju application-storage mysql database=100G,rootfs,
+`
+)
+
+var applicationStorageConfigBase = config.ConfigCommandBase{
+	Resettable: false,
+}
+
+// NewStorageCommand wraps configCommand with sane application settings.
+func NewStorageCommand() cmd.Command {
+	return modelcmd.Wrap(&storageConfigCommand{configBase: applicationStorageConfigBase})
+}
+
+type storageConfigCommand struct {
+	modelcmd.ModelCommandBase
+	configBase config.ConfigCommandBase
+	out        cmd.Output
+
+	api             StorageConstraintsAPI
+	applicationName string
+}
+
+// Info implements part of the cmd.Command interface.
```

Comment:
I realise that this is a standard that you have probably seen in a few places when understanding how to add Juju commands but we are trying to not do this type of commenting any more as it is extremely pointless.

Can we update the comment to tell us the post of the function? For example I would do something like:

```
Info returns the important usage and name info for the application storage command guiding a user on how to use the command. This func satisfies the [cmd.Command] interface.
```

Reply:
Sure will refactor this into something better, thanks tom